src/tasks/tasks.py

```diff
@@ -1389,6 +1393,86 @@ def collect_metrics(ix_to_tag_dict, tag_group):
         return collected_metrics
 
 
+@register_task("winogender-diagnostic", rel_path="RTE/diagnostics", n_classes=2)
+class WinogenderTask(GLUEDiagnosticTask):
+    """Supported wnogender task """
+
+    def __init__(self, path, max_seq_len, name, n_classes, **kw):
+        super().__init__(path, max_seq_len, name, n_classes, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+        self.n_classes = n_classes
+
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data = None
+        self.acc_scorer = BooleanAccuracy()
+        self.gender_parity_scorer = GenderParity()
```

Comment:
do these `scorers` need to be assigned as `self.scorers` so that `self.update_metrics()` will work with them?

Reply:
Not in this case because diagnostic tasks don't call update_metrics, they call update_diagnostic_metrics()